---
title: "Untitled"
output: html_document
date: '2022-04-25'
editor_options: 
  chunk_output_type: console
---

Read in libraries:
```{r}
library(dada2)
packageVersion("dada2")
library(ShortRead)
packageVersion("ShortRead")
library(Biostrings)
packageVersion("Biostrings")

path <- "~/vol1/all/"
```

Path to files: 
```{r}
projects <- c("PRJNA1013140", "PRJNA799831", "PRJNA800628", "PRJNA847791", "PRJNA850658", "PRJNA892439")

for (i in projects){
  assign(paste0("path_", i), paste0(path, i))
  
  #Sort into forward and reverse:
  assign(paste0("fnFs_", i), sort(list.files(eval(parse(text=paste0("path_", i))), pattern = "_1.fastq.gz", full.names = TRUE)))
  assign(paste0("fnRs_", i), sort(list.files(eval(parse(text=paste0("path_", i))), pattern = "_2.fastq.gz", full.names = TRUE)))
}
```

Filter out those with N in the sequence:
```{r}
for (i in projects){
  assign(paste0("fnFs.filtN_", i), file.path(eval(parse(text=paste0("path_", i))), "filtN", basename(eval(parse(text=paste0("fnFs_", i))))))
  assign(paste0("fnRs.filtN_", i), file.path(eval(parse(text=paste0("path_", i))), "filtN", basename(eval(parse(text=paste0("fnRs_", i))))))
  filterAndTrim(eval(parse(text=paste0("fnFs_", i))), eval(parse(text=paste0("fnFs.filtN_", i))), eval(parse(text=paste0("fnRs_", i))), eval(parse(text=paste0("fnRs.filtN_", i))), maxN = 0, multithread = TRUE)
}

```

Investigate hits function:
```{r}
primerHits <- function(primer, fn) {
    # Counts number of reads in which the primer is found
    nhits <- vcountPattern(primer, sread(readFastq(fn)), fixed = FALSE)
    return(sum(nhits > 0))
}
```


Project PRJNA1013140:
```{r}
#Primers:
FWD1 <- "CCTAC"
REV1 <- "TAC"


#Investigate hits:
rbind(FWD1 = sapply(FWD1, primerHits, fn = fnFs.filtN_PRJNA1013140[[1]]), 
      REV1 = sapply(REV1, primerHits, fn = fnRs.filtN_PRJNA1013140[[1]]))

#Remove primers!
system("/usr/bin/cutadapt --version")

path.cut_PRJNA1013140 <- file.path(path_PRJNA1013140, "cutadapt")
if(!dir.exists(path.cut_PRJNA1013140)) dir.create(path.cut_PRJNA1013140)
fnFs.cut_PRJNA1013140 <- file.path(path.cut_PRJNA1013140, basename(fnFs_PRJNA1013140))
fnRs.cut_PRJNA1013140 <- file.path(path.cut_PRJNA1013140, basename(fnRs_PRJNA1013140))

# Trim FWD:
R1.flags <- paste("-g", FWD1) 
# Trim REV:
R2.flags <- paste("-G", REV1) 

# Run Cutadapt
for(i in seq_along(fnFs_PRJNA1013140)) {
  system(paste("/usr/bin/cutadapt", R1.flags, R2.flags, "-n", 2, "-o", fnFs.cut_PRJNA1013140[i], "-p", fnRs.cut_PRJNA1013140[i], fnFs.filtN_PRJNA1013140[i], fnRs.filtN_PRJNA1013140[i]))
}
```


Project PRJNA799831:
```{r}
#Primers:
FWD1 <- "CTAC"
REV1 <- "TAC"

#Investigate hits:
rbind(FWD1 = sapply(FWD1, primerHits, fn = fnFs.filtN_PRJNA799831[[1]]), 
      REV1 = sapply(REV1, primerHits, fn = fnRs.filtN_PRJNA799831[[1]]))

#Remove primers!
path.cut_PRJNA799831 <- file.path(path_PRJNA799831, "cutadapt")
if(!dir.exists(path.cut_PRJNA799831)) dir.create(path.cut_PRJNA799831)
fnFs.cut_PRJNA799831 <- file.path(path.cut_PRJNA799831, basename(fnFs_PRJNA799831))
fnRs.cut_PRJNA799831 <- file.path(path.cut_PRJNA799831, basename(fnRs_PRJNA799831))

# Trim FWD:
R1.flags <- paste("-g", FWD1) 
# Trim REV:
R2.flags <- paste("-G", REV1) 

# Run Cutadapt
for(i in seq_along(fnFs_PRJNA799831)) {
    system(paste("/usr/bin/cutadapt", R1.flags, R2.flags, "-n", 2, "-o", fnFs.cut_PRJNA799831[i], "-p", fnRs.cut_PRJNA799831[i], fnFs.filtN_PRJNA799831[i], fnRs.filtN_PRJNA799831[i]))
}
```

Project PRJNA800628:
```{r}
#Primers:
FWD1 <- "CAGCAG"
REV1 <- "TAC"

#Investigate hits:
rbind(FWD1 = sapply(FWD1, primerHits, fn = fnFs.filtN_PRJNA800628[[1]]),
      REV1 = sapply(REV1, primerHits, fn = fnRs.filtN_PRJNA800628[[1]]))

#Remove primers!
path.cut_PRJNA800628 <- file.path(path_PRJNA800628, "cutadapt")
if(!dir.exists(path.cut_PRJNA800628)) dir.create(path.cut_PRJNA800628)
fnFs.cut_PRJNA800628 <- file.path(path.cut_PRJNA800628, basename(fnFs_PRJNA800628))
fnRs.cut_PRJNA800628 <- file.path(path.cut_PRJNA800628, basename(fnRs_PRJNA800628))

# Trim FWD:
R1.flags <- paste("-g", FWD1) 
# Trim REV:
R2.flags <- paste("-G", REV1) 

# Run Cutadapt
for(i in seq_along(fnFs_PRJNA800628)) {
    system(paste("/usr/bin/cutadapt", R1.flags, R2.flags, "-n", 2, "-o", fnFs.cut_PRJNA800628[i], "-p", fnRs.cut_PRJNA800628[i], fnFs.filtN_PRJNA800628[i], fnRs.filtN_PRJNA800628[i]))
}
```

Project PRJNA847791:
```{r}
#Primers:
FWD1 <- "CCTA"

REV1 <- "TAC"


#Investigate hits:
rbind(FWD1 = sapply(FWD1, primerHits, fn = fnFs.filtN_PRJNA847791[[1]]), 
      REV1 = sapply(REV1, primerHits, fn = fnRs.filtN_PRJNA847791[[1]]))

#Remove primers!
path.cut_PRJNA847791 <- file.path(path_PRJNA847791, "cutadapt")
if(!dir.exists(path.cut_PRJNA847791)) dir.create(path.cut_PRJNA847791)
fnFs.cut_PRJNA847791 <- file.path(path.cut_PRJNA847791, basename(fnFs_PRJNA847791))
fnRs.cut_PRJNA847791 <- file.path(path.cut_PRJNA847791, basename(fnRs_PRJNA847791))

# Trim FWD:
R1.flags <- paste("-g", FWD1) 
# Trim REV:
R2.flags <- paste("-G", REV1) 

# Run Cutadapt
for(i in seq_along(fnFs_PRJNA847791)) {
    system(paste("/usr/bin/cutadapt", R1.flags, R2.flags, "-n", 2, "-o", fnFs.cut_PRJNA847791[i], "-p", fnRs.cut_PRJNA847791[i], fnFs.filtN_PRJNA847791[i], fnRs.filtN_PRJNA847791[i]))
}
```

Project PRJNA850658:
```{r}
#Primers:
FWD1 <- "CCTA"

REV1 <- "TAC"

#Investigate hits:
rbind(FWD1 = sapply(FWD1, primerHits, fn = fnFs.filtN_PRJNA850658[[1]]), 
      REV1 = sapply(REV1, primerHits, fn = fnRs.filtN_PRJNA850658[[1]]))

#Remove primers!
path.cut_PRJNA850658 <- file.path(path_PRJNA850658, "cutadapt")
if(!dir.exists(path.cut_PRJNA850658)) dir.create(path.cut_PRJNA850658)
fnFs.cut_PRJNA850658 <- file.path(path.cut_PRJNA850658, basename(fnFs_PRJNA850658))
fnRs.cut_PRJNA850658 <- file.path(path.cut_PRJNA850658, basename(fnRs_PRJNA850658))

# Trim FWD:
R1.flags <- paste("-g", FWD1) 
# Trim REV:
R2.flags <- paste("-G", REV1) 

# Run Cutadapt
for(i in seq_along(fnFs_PRJNA850658)) {
    system(paste("/usr/bin/cutadapt", R1.flags, R2.flags, "-n", 2, "-o", fnFs.cut_PRJNA850658[i], "-p", fnRs.cut_PRJNA850658[i], fnFs.filtN_PRJNA850658[i], fnRs.filtN_PRJNA850658[i]))
}
```

Project PRJNA892439 (_1 -> _2?)
```{r}
#Primers:
FWD1 <- "GCT"

REV1 <- "CCTA"


#Investigate hits:
rbind(FWD1 = sapply(FWD1, primerHits, fn = fnFs.filtN_PRJNA850658[[1]]), 
      REV1 = sapply(REV1, primerHits, fn = fnRs.filtN_PRJNA892439[[1]]))

#Remove primers!
path.cut_PRJNA892439 <- file.path(path_PRJNA892439, "cutadapt")
if(!dir.exists(path.cut_PRJNA892439)) dir.create(path.cut_PRJNA892439)
fnFs.cut_PRJNA892439 <- file.path(path.cut_PRJNA892439, basename(fnFs_PRJNA892439))
fnRs.cut_PRJNA892439 <- file.path(path.cut_PRJNA892439, basename(fnRs_PRJNA892439))

# Trim FWD:
R1.flags <- paste("-g", FWD1) 
# Trim REV:
R2.flags <- paste("-G", REV1) 

# Run Cutadapt
for(i in seq_along(fnFs_PRJNA892439)) {
    system(paste("/usr/bin/cutadapt", R1.flags, R2.flags, "-n", 2, "-o", fnFs.cut_PRJNA892439[i], "-p", fnRs.cut_PRJNA892439[i], fnFs.filtN_PRJNA892439[i], fnRs.filtN_PRJNA892439[i]))
}
```